### ISSUE TITLE
Higher-order gradients

### ISSUE THREAD

**Issue by ddrous** [sail-sg/autofd#11]
Hi,

First of all, thanks for such a wonderful project !

In the ICLR paper attached to this project, you mentioned the need for gradients not only of functionals, but also for operators. 

Is there currently support for second- and third order gradients ?  If not, is there a plan to support this in the future ? Finally are you welcoming contributions ?

I'm ultimately interest in [truncated Taylor polynomials](https://jax.readthedocs.io/en/latest/jax.experimental.jet.html) for functionals in order to replace the current parametrisation->vector-valued differentials I'm using.

Thank you,
